ocaml-lsp-server/src/fmt.ml

```diff
@@ -126,7 +126,182 @@ let exec state bin args stdin =
   | Unix.WEXITED 0 -> Result.Ok res.stdout
   | _ -> Result.Error (Unexpected_result { message = res.stderr })
 
-let run state doc : (string, error) result Fiber.t =
+let insert ~line_sep ~line added_lines =
+  let start = { Position.character = 0; line } in
+  let end_ = { Position.character = 0; line } in
+  let range = { Range.start; end_ } in
+  let newText =
+    (added_lines |> Array.to_list |> String.concat ~sep:line_sep) ^ line_sep
+  in
+  { TextEdit.newText; range }
+
+let replace ~line_sep ~line deleted_lines added_lines =
```

Comment:
`insert`, `replace`, and `delete` are quite similar. Can you combine them somehow?

```
type edit =
   | Add of ..
   | Replace of ..
   | Delete of ..
val text_edit : line:int -> line_sep:string -> edit -> TextEdit.t
```